src/Network/Transport/Internal.hs

```diff
@@ -68,42 +73,66 @@ foreign import ccall unsafe "ntohs" ntohs :: CShort -> CShort
 #endif
 
 -- | Serialize 32-bit to network byte order
-encodeInt32 :: Enum a => a -> ByteString
-encodeInt32 i32 =
+encodeWord32 :: Word32 -> ByteString
+encodeWord32 i32 =
   BSI.unsafeCreate 4 $ \p ->
-    pokeByteOff p 0 (htonl . fromIntegral . fromEnum $ i32)
+    -- fromIntegral :: Word32 -> CInt
```

Comment:
I see that in this hackage those functions are inlined and are using CInt type, but what is the point, according to POSIX.1-2001/2008 the family of those functions are:
```
       uint32_t htonl(uint32_t hostlong);
       uint16_t htons(uint16_t hostshort);
       uint32_t ntohl(uint32_t netlong);
       uint16_t ntohs(uint16_t netshort);
```

So either `CUInt` or better `WordN` should be used here? Then no conversion is needed.